parquet-column/src/main/java/org/apache/parquet/column/statistics/BinaryStatistics.java

```diff
@@ -46,8 +46,8 @@ public void mergeStatisticsMinMax(Statistics stats) {
 
   @Override
   public void setMinMaxFromBytes(byte[] minBytes, byte[] maxBytes) {
-    max = Binary.fromByteArray(maxBytes);
-    min = Binary.fromByteArray(minBytes);
+    max = Binary.fromUnmodifiedByteArray(maxBytes);
+    min = Binary.fromUnmodifiedByteArray(minBytes);
```

Comment:
I think this should either use the "reused" call or require the caller wrap the byte arrays. That way there isn't a way for future callers to use this unsafely.